components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1438,60 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, Unit<Length> unit) {
+    if (value == null || value.isInfinite()) {
+      LOGGER.debug("Expected float value for position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) unit = UNITS.REFERENCEFRAME;
```

Comment:
I don't think `getPosition(Double)` makes sense to have.  I think we should require the caller to specify the unit in all cases, even when that unit is "reference frame" so as to be completely unambiguous.

In the case that the value *or* the unit are `null`, I think we should be simply returning `null` or throwing an exception here, so that nulls are propagated through or are required to be handled.

The main benefit I see in this PR is the ability to create a `Length` using either the enum value or a string.  But I wonder if this couldn't be added at a lower level--as a factory method for the ome-xml Quantity class, for example, to make use of it simple and direct.  And it would also make it work with all unit types rather than just lengths.